File: 2022/day1/part1.rs
use std::io;
use std::cmp;

fn main() {
    let mut max = 0;
    let mut elf = 0;

    for line in io::stdin().lines() {
        match line.unwrap().parse::<u32>() {
            Ok(n) => {
                elf += n
            },
            Err(_) => {
                max = cmp::max(max, elf);
                elf = 0;
            },
        };
    }

    max = cmp::max(max, elf);
    println!("{max}");
}

File: 2022/day1/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	max := 0
	elf := 0

	for s := bufio.NewScanner(os.Stdin); s.Scan(); {
		if n, err := strconv.Atoi(s.Text()); err == nil {
			elf += n
		} else {
			max = Max(max, elf)
			elf = 0
		}
	}

	max = Max(max, elf)
	fmt.Println(max)
}

func Max(x, y int) int {
	if x > y {
		return x
	}
	return y
}
